### PR TITLE
GUACAMOLE-527: Client-side changes for SSH host key checking

### PR DESCRIPTION
**Pull request by necouchman**
The client-side changes for passing through a known host key to guacd.

### CODE REVIEW

**Comment by necouchman** [comment]
Merge base updated to staging/1.0.0

**Comment by necouchman** [comment]
@mike-jumper - ping on this one since it is the matching change to the recently-merged apache/guacamole-server#164

File: guacamole-ext/src/main/resources/org/apache/guacamole/protocols/rdp.json [hunk 1]
```diff
@@ -329,6 +329,10 @@
                     "name"  : "sftp-hostname",
                     "type"  : "TEXT"
                 },
+                {
+                    "name"  : "sftp-host-key",
```
Comment (mike-jumper) [inline guacamole-ext/src/main/resources/org/apache/guacamole/protocols/rdp.json]: Is there a reason behind the difference in parameter order across protocols? I see `hostname`/`port`/**`host-key`** for SSH but `sftp-hostname`/**`sftp-host-key`**/`sftp-port` for VNC/RDP.

File: guacamole-ext/src/main/resources/org/apache/guacamole/protocols/ssh.json [hunk 2]
```diff
@@ -12,6 +12,10 @@
                 {
                     "name"  : "port",
                     "type"  : "NUMERIC"
+                },
+                {
+                    "name"  : "host-key",
+                    "type"  : "MULTILINE"
```
Comment (mike-jumper) [inline guacamole-ext/src/main/resources/org/apache/guacamole/protocols/ssh.json]: Does this value contain multiple lines? I thought each host was defined by a single line.

File: guacamole/src/main/webapp/translations/en.json [hunk 3]
```diff
@@ -364,6 +364,7 @@
         "FIELD_HEADER_SECURITY"        : "Security mode:",
         "FIELD_HEADER_SERVER_LAYOUT"   : "Keyboard layout:",
         "FIELD_HEADER_SFTP_DIRECTORY"             : "Default upload directory:",
+        "FIELD_HEADER_SFTP_HOST_KEY"              : "SSH host key (Base64):",
```
Comment (mike-jumper) [inline guacamole/src/main/webapp/translations/en.json]: As the fact this is the key of the SSH host is given from the header for the section (SFTP), but the public vs. private nature of that key is not, I suggest using the same text for VNC/RDP that you use for the SSH parameter ("public host key").

File: guacamole-ext/src/main/resources/org/apache/guacamole/protocols/rdp.json (hunk 1 above)
Comment (necouchman) [reply guacamole-ext/src/main/resources/org/apache/guacamole/protocols/rdp.json]: No, just a mistake.  Reordered.

File: guacamole-ext/src/main/resources/org/apache/guacamole/protocols/ssh.json (hunk 2 above)
Comment (necouchman) [reply guacamole-ext/src/main/resources/org/apache/guacamole/protocols/ssh.json]: Yeah, I think you're right.

Comment (necouchman) [reply guacamole-ext/src/main/resources/org/apache/guacamole/protocols/ssh.json]: Fixed.

File: guacamole/src/main/webapp/translations/en.json (hunk 3 above)
Comment (necouchman) [reply guacamole/src/main/webapp/translations/en.json]: Yeah, good point.  Fixed.

**Pull request merged**